sw.js

```diff
@@ -1,103 +1,85 @@
-  cacheName
-});
+const CACHES = {
+  cssJs: 'css-js',
+  images: 'images',
+  fonts: 'google-fonts'
+};
 
-/**
- * Route for local CSS and JS assets
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#stale-while-revalidate
- */
-const assetRoute = new routing.RegExpRoute({
-  regExp: new RegExp(`^${localhost}.*\\.(css|js|png|svg)$`),
-  handler: new runtimeCaching.StaleWhileRevalidate({ requestWrapper })
-});
+// Cache fallback for HTML files
+workbox.routing.registerRoute(/(?:^[^.]+|\.html)$/, args =>
+  workbox.strategies
+    .networkFirst()
+    .handle(args)
+    .then(
+      // page is not in cache, respond with 404 page (even if the page exists)
+      response => (response === undefined ? caches.match('/404/') : response)
+    )
+);
 
-/**
- * Route for assets on cloudfront.net CDN
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#stale-while-revalidate
- */
-const cdnAssetRoute = new routing.RegExpRoute({
-  regExp: new RegExp('^https://.*\\.cloudfront\\.net/.*\\.(png|svg)$'),
-  handler: new runtimeCaching.StaleWhileRevalidate({
-    requestWrapper: cdnRequestWrapper
+// Cache CSS and JS files
+workbox.routing.registerRoute(
+  /\.(?:js|css)$/,
+  workbox.strategies.staleWhileRevalidate({
+    cacheName: CACHES.cssJs
   })
-});
-
-/**
- * Route for pages
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#network-falling-back-to-cache
- */
-const navRoute = new routing.NavigationRoute({
-  whitelist: [/./],
-  handler: new runtimeCaching.NetworkFirst({ requestWrapper })
-});
-
-/**
- * Precache resources
- */
-staticCache.addToCacheList({
-  unrevisionedFiles: ['/', '/404/', '/error/']
-});
+);
 
-self.oninstall = event => {
-  event.waitUntil(Promise.all([staticCache.install().then(skipWaiting)]));
-};
+// Cache images
+workbox.routing.registerRoute(
+  /\.(?:png|gif|jpg|jpeg|svg)$/,
+  workbox.strategies.cacheFirst({
+    cacheName: CACHES.images,
+    plugins: [
+      new workbox.expiration.Plugin({
+        maxEntries: 60,
+        maxAgeSeconds: 30 * 24 * 60 * 60 // 30 Days
+      })
+    ]
+  })
+);
 
-self.onactivate = event => {
-  caches.keys().then(keys => {
-    keys.forEach(cacheName => {
-      if (!cacheName.endsWith(version)) {
-        caches.delete(cacheName);
-      }
-    });
-  });
-  return event.waitUntil(clients.claim());
-};
+// Cache Google Fonts
+workbox.routing.registerRoute(
+  new RegExp('https://fonts.(?:googleapis|gstatic).com/(.*)'),
+  workbox.strategies.cacheFirst({
+    cacheName: CACHES.fonts,
+    plugins: [
+      new workbox.expiration.Plugin({
+        maxEntries: 30
+      }),
+      new workbox.cacheableResponse.Plugin({
+        statuses: [200]
+      })
+    ]
+  })
+);
 
-router.registerRoutes({
-  routes: [assetRoute, cdnAssetRoute, navRoute]
-});
+workbox.precaching.precache(['/', '/404/', '/error/']);
```

Comment:
@calebeby Hmm...I'm getting a Workbox warning. 🤔 

> The following precache entries might not be revisioned:
>
>    - "/"
>     - "/404/"
>     - "/error/"
> 
> You can learn more about why this might be a problem here: https://developers.google.com/web/tools/workbox/modules/workbox-precaching

Maybe you can look into using the Workbox CLI (maybe as an npm task?) to automate this for us?

Reply:
Should I do that in this PR or another?

Reply:
@calebeby I think it should be a part of this PR.

Reply:
On it!

Reply:
Also, if you have a network connection, and the server responds with a 404, you can just respond with the 404 page that came from the server, instead of doing caching things with that